src/backend/distributed/commands/extension.c

```diff
@@ -115,3 +110,516 @@ ExtractNewExtensionVersion(Node *parsetree)
 
 	return newVersion;
 }
+
+
+/*
+ * PlanCreateExtensionStmt is called during the creation of an extension.
+ * It is executed before the statement is applied locally.
+ * We decide if the extension needs to be replicated to the worker, and
+ * if that is the case return a list of DDLJob's that describe how and
+ * where the extension needs to be created.
+ */
+List *
+PlanCreateExtensionStmt(CreateExtensionStmt *createExtensionStmt, const char *queryString)
+{
+	List *commands = NIL;
+	const char *createExtensionStmtSql = NULL;
+
+	if (!ShouldPropagateExtensionCommand((Node *) createExtensionStmt))
+	{
+		return NIL;
+	}
+
+	/* Extension management can only be done via coordinator node */
+	EnsureCoordinator();
+
+	/*
+	 * Here we append "new_version" and "schema" fields to the "options" list
```

Comment:
Let's not forget to removing the code for always adding the version and updating the comment. 

In that case, it might be good to rename the function to `AddSchemaFieldIfMissing(createExtensionStmt)` or such